src/main/java/com/autotune/analyzer/services/UpdateResults.java

```diff
@@ -67,19 +68,24 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
         Utils.addCORSHeaders(response);
         try {
             String inputData = request.getReader().lines().collect(Collectors.joining());
-            List<ExperimentResultData> experimentResultDataList = new ArrayList<ExperimentResultData>();
+            List<ExperimentResultData> experimentResultDataList = new ArrayList<>();
             List<UpdateResultsSO> updateResultsSOList = Arrays.asList(new Gson().fromJson(inputData, UpdateResultsSO[].class));
-            for (UpdateResultsSO updateResultsSO : updateResultsSOList) {
-                experimentResultDataList.add(Utils.Converters.KruizeObjectConverters.convertUpdateResultsSOToExperimentResultData(updateResultsSO));
-            }
-            LOGGER.debug(experimentResultDataList.toString());
-            new ExperimentInitiator().validateAndUpdateResults(mainKruizeExperimentMap, experimentResultDataList, performanceProfilesMap);
-            ExperimentResultData invalidKExperimentResultData = experimentResultDataList.stream().filter((rData) -> (!rData.getValidationResultData().isSuccess())).findAny().orElse(null);
-            if (null == invalidKExperimentResultData) {
-                sendSuccessResponse(response, "Results added successfully! View saved results at /listExperiments.");
+            if (updateResultsSOList.size() > 1) {
+                LOGGER.error(AnalyzerErrorConstants.AutotuneObjectErrors.UNSUPPORTED_EXPERIMENT);
+                sendErrorResponse(response, null, HttpServletResponse.SC_BAD_REQUEST, AnalyzerErrorConstants.AutotuneObjectErrors.UNSUPPORTED_EXPERIMENT );
             } else {
-                LOGGER.error("Unable to save results due to :" + invalidKExperimentResultData.getValidationResultData().getMessage());
-                sendErrorResponse(response, null, HttpServletResponse.SC_BAD_REQUEST, invalidKExperimentResultData.getValidationResultData().getMessage());
+                for (UpdateResultsSO updateResultsSO : updateResultsSOList) {
+                    experimentResultDataList.add(Utils.Converters.KruizeObjectConverters.convertUpdateResultsSOToExperimentResultData(updateResultsSO));
+                }
+                LOGGER.debug(experimentResultDataList.toString());
+                new ExperimentInitiator().validateAndUpdateResults(mainKruizeExperimentMap, experimentResultDataList, performanceProfilesMap);
+                ExperimentResultData invalidKExperimentResultData = experimentResultDataList.stream().filter((rData) -> (!rData.getValidationResultData().isSuccess())).findAny().orElse(null);
+                if (null == invalidKExperimentResultData) {
+                    sendSuccessResponse(response, "Results added successfully! View saved results at /listExperiments.");
+                } else {
+                    LOGGER.error("Unable to save results due to :" + invalidKExperimentResultData.getValidationResultData().getMessage());
```

Comment:
Remove `due to` in all places, it is redundant.
Should be `Failed to update results: `, to be consistent